app/controllers/api_controller/parser/request_adapter.rb

```diff
@@ -20,7 +20,7 @@ def base
       end
 
       def collection
-        @params[:collection]
+        path.split("/")[version_override? ? 3 : 2]
```

Comment:
maybe memoize path.split("/") for collection and subcollection, the others are.  Maybe a path_parts method.